blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -138,9 +141,14 @@ class RunDetails extends Component {
     afterClose = () => {
         const { router, params } = this.context;
         if (this.opener) {
-            // step back in the history to the item that's prior to initial load of RunDetails
-            const offsetToInitialRoute = this.initialHistoryLength - history.length;
+            // move back in the history to the item that's prior to initial load of RunDetails
+            const offsetToInitialRoute = this.initialHistoryLength - locationService.navCount;
             router.go(offsetToInitialRoute - 1);
+            setTimeout(() => {
+                if (locationService.current !== this.opener) {
+                    router.replace(this.opener);
```

Comment:
Why `replace`? Especially because we first go to a specific history `router.go(offsetToInitialRoute - 1)` and then change that history entry.

Reply:
It's a fail-safe. There was a bug that was causing this code to fail because the browser seems to be capping its actual history to 50, so when the page was first entered it was initially 50, then if the user clicked any of the tabs to navigate, it was still 50 and it would only go back 1 to an incorrect url. This is why I added tracking at the topmost router level. It's possible that could miss navigations, so this fail safe restores the correct url in the case it's not right.

Reply:
FYI the result of the incorrect URL caused excessive full page reloads when doing anything with any tabs after the browser history got too large.

Reply:
I know some of the code here was to handle a couple of bugs where closing run details didn't update the route correctly, resulting in the UI not actually re-rendering and displaying the cloned UI underneath (which caused that full page refresh after click). We just want to keep an eye on ATH, and maybe see if there's anything we could add there that would give us more confidence in this change.